tests/fixtures/scope-analysis/class-supper-type.ts

```diff
@@ -0,0 +1,11 @@
+abstract class Foo extends Bar<Baz> {
```

Comment:
I'm guessing the `Bar<Baz>` should be a `TSTypeReference` node...

Reply:
`<Baz>` is TSTypeReference

Reply:
https://gist.github.com/armano2/b26b9e77438c72c5b1c875024272c2ed

Reply:
superTypeParameters -> TSTypeParameterInstantiation[params] -> TSTypeReference

Reply:
`Bar` is `superClass` -> `Identifier`

Reply:
Mmm. It's not ideal...

```ts
class Foo1 extends T<U> {}
class Foo2 extends Wrap<T<U>> {}
function f(a: T<U>): T<U> {}
const A: T<U> = {}
```

Every `T<U>` should be:

```jsonc
{
  "type": "TSTypeReference", // T<U>
  "typeName": {
    "type": "Identifier", // T
    "name": "T"
  },
  "typeParameters": {
    "type": "TSTypeParameterInstantiation", // <U>
    "params": [
      {
        "type": "TSTypeReference", // U
        "typeName": {
          "type": "Identifier",
          "name": "U"
        }
      }
    ]
  }
```

But the first one is different...

Reply:
Ah... I see, because the `T` of the `extends T` is a variable rather than a type.
Mmmm.

Reply:
OK, fair enough.

Reply:
`extends T` is in superClass